Core/CharacterVariable.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace PCSharpGen.Core
+{
+    public class CharacterVariable
+    {
+        private readonly Character _character;
+
+        private readonly List<GameValue> _values;
```

Comment:
I think `GameValue` is a bit vauge. These are "bonuses" right?

Reply:
Bonuses implies a positive value.  I guess they are... modifiers?  But that's a weird name... "VariableModifier" maybe?

Reply:
At least in Pathfinder, there are "penalties" and "bonuses", which, combined, are "modifiers".  Since we generally won't know the sign of the value, we can't call it either penalty or bonus.  And modifier looks weird in code (since it tends to mean a lot of other things). That's why I ended up with the stupidly vague name I came up with.  It _is_ a horribly vague concept... it's literally any numeric value that would apply to any variable anywhere. :-)

Reply:
Technically a Bonus can be negative, pcgen calls them that even if they are negative, i'd be more happy with "modifier" than "value" but that means other stuff in d20 land.

Reply:
A bonus cannot be negative. It's an important distinction in rules text. (For example, when you are flat footed, you lose your Dex BONUS to AC, but if it's a penalty, you don't lose the penalty).  Also, penalties always stack, and bonuses to not.  The rules texts don't usually have these problems because it's very, very rare that an ability doesn't know whether something is positive or negative (or they give it a "min 0" clause to prevent a negative bonus).

There really isn't a great game term that encompass both, because the only time it's maybe negative is when it's coming from an ability score, when it's called modifier (which is then either a bonus).

I change it over to modifier for now.

Reply:
Gah, the word modifier by itself is a pain... And "GameModifier" is even weirder, because it sounds like it's modifying something.  It could be super weirdly verbose and be "BonusOrPenalty".  Or maybe we just accept the incorrect term "Bonus" in the code as "close enough".

Currently I think I like BonusOrPenalty or just Bonus.  I'm going to mull it over for a bit.

Reply:
I went with just "bonus", with a big comment explaining that it's really bonus/penalty, but penalties will have a negative value.

Reply:
Yea, maybe AttributeModifier or something. I'd be okay with BonusOrPenalty though.